test/unit/Chainweb/Test/Pact5/RemotePactTest.hs

```diff
@@ -692,6 +695,104 @@ gasPurchaseFailureMessages rdb _step = runResourceT $ do
+        advanceAllChains_ fx
+        checkPactVersion fx v cid >>= P.equals Pact5
+
+-- | Test that xchains work across the Pact4->Pact4 transition boundary.
+--   This is mostly the same as 'spvTest', except it waits for the transition.
+transitionCrosschain :: RocksDb -> Step -> IO ()
+transitionCrosschain rdb step = runResourceT $ do
+    let v = instantCpmTransitionTestVersion petersonChainGraph
+    let srcChain = unsafeChainId 0
+    let targetChain = unsafeChainId 9
+    fx <- mkFixture v rdb
+
+    let checkIsVersion pv = do
+            checkPactVersion fx v srcChain >>= P.equals pv
+            checkPactVersion fx v targetChain >>= P.equals pv
+
+    liftIO $ do
+        checkIsVersion Pact4
+
+        initiator <- buildTextCmd v
+            $ set cbSigners
+                [ mkEd25519Signer' sender00
+                    [ CapToken (QualifiedName "GAS" (ModuleName "coin" Nothing)) []
+                    , CapToken (QualifiedName "TRANSFER_XCHAIN" (ModuleName "coin" Nothing))
+                        [ PString "sender00"
+                        , PString "sender01"
+                        , PDecimal 1.0
+                        , PString (chainIdToText targetChain)
+                        ]
+                    ]
+                ]
+            $ set cbRPC (mkExec ("(coin.transfer-crosschain \"sender00\" \"sender01\" (read-keyset 'k) \"" <> chainIdToText targetChain <> "\" 1.0)") (mkKeySetData "k" [sender01]))
+            $ defaultCmd srcChain
+
+        step "xchain initiate"
+        send fx v srcChain [initiator]
+        let initiatorReqKey = cmdToRequestKey initiator
+        (sendCut, _) <- advanceAllChains fx
+        [Just sendCr] <- pollWithDepth fx v srcChain [initiatorReqKey] (Just (ConfirmationDepth 0))
+        let cont = fromMaybe (error "missing continuation") (_crContinuation sendCr)
+
+        step "waiting until pact5 transition"
+
+        step "... performing transition"
+        replicateM_ 16 $ advanceAllChains_ fx
+        checkIsVersion Pact4
+        advanceAllChains_ fx
+        checkIsVersion Pact5
+
+        let sendHeight = sendCut ^?! ixg srcChain . blockHeight
+        spvProof <- createTransactionOutputProof_ (fx ^. to _cutFixture . CutFixture.fixtureWebBlockHeaderDb) (fx ^. to _cutFixture . CutFixture.fixturePayloadDb) targetChain srcChain sendHeight 0
+        let contMsg = ContMsg
+                { _cmPactId = _peDefPactId cont
+                , _cmStep = succ $ _peStep cont
+                , _cmRollback = _peStepHasRollback cont
+                , _cmData = PUnit
+                , _cmProof = Just (ContProof (B64U.encode (BL.toStrict (A.encode spvProof))))
+                }
+        step "xchain recv"
+
+        recv <- buildTextCmd v
+            $ set cbRPC (mkCont contMsg)
+            $ defaultCmd targetChain
+        send fx v targetChain [recv]
+        let recvReqKey = cmdToRequestKey recv
+        advanceAllChains_ fx
+        [Just recvCr] <- poll fx v targetChain [recvReqKey]
```

Comment:
I like to do
```hs
poll fx v targetChain [recvReqKey] >>= P.list
  [ P.match _Just
    ? P.checkAll 
      [ ...
      ]
  ]
```
thus fitting it all into the property paradigm, but it's not really a big deal.